tests/core/tax_scales/test_abstract_rate_tax_scale.py

```diff
@@ -0,0 +1,71 @@
+from numpy import array
+
+from openfisca_core.taxscales import AbstractRateTaxScale
+from openfisca_core.tools import assert_near
+
+
+def test_bracket_indices():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base)
+
+    assert_near(result, [0, 0, 0, 1, 1, 2])
+
+
+def test_bracket_indices_with_factor():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base, factor = 2.0)
+
+    assert_near(result, [0, 0, 0, 0, 1, 1])
+
+
+def test_bracket_indices_with_round_decimals():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base, round_decimals = 0)
+
+    assert_near(result, [0, 0, 0, 1, 1, 2])
+
+
+def test_bracket_indices_without_tax_base():
+    tax_base = array([])
+    tax_scale = AbstractRateTaxScale()
+    tax_scale.add_bracket(0, 0)
+    tax_scale.add_bracket(2, 0)
+    tax_scale.add_bracket(4, 0)
+
+    result = tax_scale.bracket_indices(tax_base)
+
+    assert_near(result, [])
+
+
+def test_bracket_indices_without_brackets():
+    tax_base = array([0, 1, 2, 3, 4, 5])
+    tax_scale = AbstractRateTaxScale()
+
+    result = tax_scale.bracket_indices(tax_base)
+
+    assert_near(result, [0, 0, 0, 0, 0, 0])
```

Comment:
@benjello I expected this test to pass, What's the expected result?

Reply:
An error: there are no brackets at all, thus Can't calculate indices.

Reply:
As you're inserting `inf`, there will be no error though.

Reply:
Yep. But we should raise an error when doing calculation on an empty TaxScale. Or being explicit about a default tax scale. But I prefer the former solution.

Reply:
Removing `inf` gives the same result.

Reply:
What's the `inf` for? Removing it has no impact, see my latest commit.

Reply:
> But we should raise an error when doing calculation on an empty TaxScale

There're two scenarios :

1. Empty `tax_base`: gives `[-1] * n`, Should it raise an error?
2. Empty `thresholds` (brackets): gives `[0] * n`, Should it raise an error?

Reply:
For the first scenario it could be:

```python
if not size(array(tax_base)):
    return SomeError(msg)
```

Reply:
I'd prefer not to:

1. Give a default scale value
2. Chance `calc`'s signature

As both will force us to do a major version bump.

Reply:
Both. Empty tax_base should raise an error an empty tax-scale (no bracket) should raise an error too.

Reply:
Could you add that @maukoquiroga , I  am not error-raising fluent ;-)

Reply:
@benjello https://github.com/openfisca/openfisca-core/pull/920/commits/149034eaef44f776b638c0e29c6c7d4ee2a816f8 ?

Reply:
OK. Should I finish the job of letting the tests pass ?

Reply:
I fix the tests but you may want to add more of them.

Reply:
Complemented them, good job.